src/sidebar/components/search-input.js

```diff
@@ -29,7 +29,11 @@ export default function SearchInput({ alwaysExpanded, query, onSearch }) {
 
   const onSubmit = e => {
     e.preventDefault();
-    onSearch(input.current.value);
+    if (input.current.value || prevQuery) {
+      // Don't set an initial empty query, but allow a later empty query to
+      // clear `prevQuery`
+      onSearch(input.current.value);
+    }
```

Comment:
Is this change something that could be separated out of this PR?

Reply:
This change has two effects:

1. Setting a search query resets the `expanded` selection state. This has the effect of folding up expanded threads as soon as a user clicks into the search input because before this change, the form would immediately submit. That's annoying: if you've got a bunch of open nested replies, they close up again as soon as you click into the input. This has been the situation for a while, and so a change for this reason _could_ be separated from this PR.
2. Setting a search query first clears the selection (including selected annotations). That means if you are viewing a set of selected annotations and then click the search icon (before these changes, when that would cause an immediate submission), all of your selected annotations are immediately cleared, even though you haven't searched for anything yet. This is relevant to these changes: this behavior didn't used to manifest, as filter queries and selected annotations would combine.

Reply:
Addendum: the real fix here is that the search icon would not double as a submit button.